file-formats/uipg/type/zif_aff_uipg_v1.intf.abap

```diff
@@ -0,0 +1,169 @@
+      "! Tile id
+      tile_id             TYPE c LENGTH 50,
+      "! <p class="shorttext">Tile Display Format</p>
+      "! tile display format
+      tile_display_format TYPE ty_tile_display_format,
+      "! <p class="shorttext">Target Mapping Catalog Id</p>
+      "! Target Mapping catalog id
+      tm_catalog_id       TYPE c LENGTH 35,
+      "! <p class="shorttext">Target Mapping Catalog Type</p>
+      "! Target Mapping catalog type
+      tm_catalog_type     TYPE ty_catalog_type,
+      "! <p class="shorttext">Target Mapping Id</p>
+      "! Target Mapping id
+      tm_id               TYPE c LENGTH 50,
+    END OF ty_tile_assignment.
+
+  "! <p class="shorttext">Item Assignment Type</p>
+  "! Item assignment type
+  "! $values {@link zif_aff_uipg_v1.data:co_item_type}
+  "! $default {@link zif_aff_uipg_v1.data:co_item_type.tile}
+  TYPES ty_item_type TYPE c LENGTH 2.
+
+  CONSTANTS:
+    "! <p class="shorttext">Item Assignment Type</p>
+    "! Item assignment type
+    BEGIN OF co_item_type,
+      "! <p class="shorttext">Tile</p>
+      "! Tile
+      "! $enumValue 'TI'
+      tile TYPE ty_item_type VALUE 'TI',
+    END OF co_item_type.
+
+  TYPES:
+    "! <p class="shorttext">Page Section Item Assignment</p>
+    "! Page Section Item Assignment
+    BEGIN OF ty_item,
+      "! <p class="shorttext">ID</p>
+      "! Id of the Assignment
+      id              TYPE c LENGTH 35,
+      "! <p class="shorttext">Type</p>
+      "! Type of the Assignment
+      type            TYPE ty_item_type,
+      "! <p class="shorttext">ID</p>
+      "! Id of the Page Section
+      tile_assignment TYPE ty_tile_assignment,
+    END OF ty_item,
+
+    ty_items TYPE STANDARD TABLE OF ty_item WITH DEFAULT KEY.
+
+  TYPES:
+    "! <p class="shorttext">Page Section</p>
+    "! Page Section
+    BEGIN OF ty_section,
+      "! <p class="shorttext">ID</p>
+      "! Id of the Page Section
+      id    TYPE c LENGTH 35,
+      "! <p class="shorttext">Title</p>
+      "! Section title
+      title TYPE c LENGTH 100,
+      "! <p class="shorttext">Items</p>
+      "! Assigned Page Section Items
+      items TYPE ty_items,
+    END OF ty_section,
+
+    ty_sections TYPE STANDARD TABLE OF ty_section WITH DEFAULT KEY.
+
+  TYPES:
+    "! <p class="shorttext">Fiori Launchpad Page Template</p>
+    "! Fiori launchpad page template
+    BEGIN OF ty_main,
+      "! <p class="shorttext">Format Version</p>
+      "! Format version
+      "! $required
+      format_version      TYPE zif_aff_types_v1=>ty_format_version,
+
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header              TYPE zif_aff_types_v1=>ty_header_100_cloud,
+
+      "! <p class="shorttext">General Information</p>
+      "! General information
+      "! $required
+      general_information TYPE ty_general_information,
+
+      "! <p class="shorttext">Page Sections</p>
+      "! Sections of the Page Template
+      sections            TYPE ty_sections,
```

Comment:
The challenge here is that our names are so incredible generic. I mean, once we have the AFF v.2 we will most likely deal with 3 different objects that are all called item.

I'll leave the Page* away, since it should be clear from the context that we are talking about Page (templates)